### ISSUE TITLE
Create rooms without clients

### ISSUE THREAD

**Issue by GreenArrow18** [colyseus/colyseus#270]
Is there a way to create rooms which are initiated by the server, instead of waiting for a client to connect? I would like to be able to automatically create some rooms when the server starts.

However, it seems like room creation can only be triggered when a client connects and sends a create request to the server?

Thanks.

**Comment by endel** [comment]
Hi @GreenArrow18,

I think you should be able to do this currently, if you do the following:

```typescript
class YourRoom extends Room {
  autoDispose = false;
  // ...
}

gameServer.define("your_room", YourRoom);
gameServer.matchMaker.createRoom("your_room", { /* options */ });
```

Let me know if that works for you, otherwise we can make the necessary changes to allow this to be possible. Cheers!

**Comment by GreenArrow18** [comment]
Yes, this works and so far seems enough for what I need it for.

Thanks!

**Comment by GreenArrow18** [comment]
Hi. I just updated to the latest version, and I am now getting this error:

> error TS2339: Property 'matchMaker' does not exist on type 'Server'

Has there been changes in this area? And how do I get previous code to work again?

Thanks.

**Comment by endel** [comment]
Hi @GreenArrow18, you can import the `matchMaker` directly now.

```typescript
import { matchMaker } from "colyseus"
```

**Comment by GreenArrow18** [comment]
Ok got it working now.

Thanks again!

**Comment by FadiBunni** [comment]
This does not seems to be working anymore.

**Comment by GreenArrow18** [comment]
@FadiBunni are you getting any error message?

Steps mentioned above still work for me.

**Comment by FadiBunni** [comment]
I got it working now. 

Yes it seems that this: autoDispose = false; must be in the room class. 

And gameServer.matchMaker.createRoom("your_room", { /* options */ }) must be in the "server.js" or "app.js". I thought they had to be togeather. 

Also for me only this works: MatchMaker.createRoom("your_room", { /* options */ }). You cannot call the gameServer and expect to find the MatchMaker property.

**Comment by xhacker5000** [comment]
after call
``matchMaker.createRoom('my_room',{create_token:'test_create_token'})``
when the client come
then error
``server.handleUpgrade() was called more than once with the same socket``

**Comment by robknopf** [comment]
Has this regressed?  I'm trying to create a room before a client has connected.  Calling matchMaker.createRoom("my_room", {}) inside beforeListen and an error is thrown.:

```
   beforeListen: async () => {
        /**
         * Before before gameServer.listen() is called.
         */
        // create a room initially, instead of waiting for a client to create one
        const existingRoom = await matchMaker.findOneRoomAvailable("my_room", { /* options */ })
        if (!existingRoom) {
            await matchMaker.createRoom("my_room", { /* options */ });
        }
    }
```

... throws an error in MatchMaker.js:96.  Looks like the selectProcessIdToCreateRoom is trying to index into an empty list to find a processId (no rooms exist yet):

```
selectProcessIdToCreateRoom = _selectProcessIdToCreateRoom || async function () {
    return (await stats.fetchAll()).sort((p1, p2) => p1.roomCount > p2.roomCount ? 1 : -1)[0].processId;
  };
```

Am I trying to create the room in the wrong place?

**Comment by endel** [comment]
Thank you for noticing @robknopf, indeed this regressed at https://github.com/colyseus/colyseus/commit/477df822e6d608584d47e0143d82a85330ee5f02, just fixed on version `0.15.26` (https://github.com/colyseus/colyseus/commit/55017667903e0526678cc28617b92fe576a11874)

**Comment by antonytrupe** [comment]
I'm getting every room twice.
